File: src/test/console/java/src/main/java/web/JavaScriptAlertsPage.java
package web;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;

public class JavaScriptAlertsPage extends Base {
  @FindBy(xpath = "//*[@id=\"content\"]/div/h3[contains(.,\"JavaScript Alerts\")]")
  public WebElement header;
  @FindBy(xpath ="//*[@id=\"content\"]/div/ul/li/button[contains(., \"Click for JS Alert\")]" )
  public WebElement clickForJSAlertButton;
  @FindBy(xpath = "//*[@id=\"content\"]/div/ul/li/button[contains(., \"Click for JS Confirm\")]")
  public WebElement clickForJSConfirmButton;
  @FindBy(xpath = "//*[@id=\"content\"]/div/ul/li/button[contains(., \"Click for JS Prompt\")]")
  public WebElement clickForJSPromptButton;
  @FindBy(xpath = "//*[@id=\"result\"]")
  public WebElement resultMessage;

  public String alertMessage = "You successfully clicked an alert";
  public String confirmWithOKMessage = "You clicked: Ok";
  public String confirmWithCancelMessage = "You clicked: Cancel";

  public JavaScriptAlertsPage(WebDriver driver, long timeoutInSeconds) {
    super(driver, timeoutInSeconds, "http://the-internet.herokuapp.com/javascript_alerts");
  }

  public void open() throws Exception {
    super.open();
    this.waitForElementVisible(header);
  }

  public void clickOnJSAlertButton() throws Exception {
    clickForJSAlertButton.click();
    this.waitForAlertVisible();
  }

  public void clickOnJSConfirmButton() throws Exception {
    clickForJSConfirmButton.click();
    this.waitForAlertVisible();
  }

  public void clickOnJSPromptButton() throws Exception {
    clickForJSPromptButton.click();
    this.waitForAlertVisible();
  }
}

File: src/test/console/dotnet/Jsonwired/web/JavascriptAlertsPage.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Support.PageObjects;

namespace web {
	public class JavaScriptAlertsPage : Base {
		[FindsBy(How = How.XPath, Using = "//*[@id=\"content\"]/div/h3[contains(.,\"JavaScript Alerts\")]")]
		public IWebElement header ;
		[FindsBy(How = How.XPath, Using = "//*[@id=\"content\"]/div/ul/li/button[contains(., \"Click for JS Alert\")]")]
		public IWebElement clickForJSAlertButton ;
		[FindsBy(How = How.XPath, Using = "//*[@id=\"content\"]/div/ul/li/button[contains(., \"Click for JS Confirm\")]")]
		public IWebElement clickForJSConfirmButton ;
		[FindsBy(How = How.XPath, Using = "//*[@id=\"content\"]/div/ul/li/button[contains(., \"Click for JS Prompt\")]")]
		public IWebElement clickForJSPromptButton ;
		[FindsBy(How = How.XPath, Using = "//*[@id=\"result\"]")]
		public IWebElement resultMessage ;

		public string alertMessage  = "You successfully clicked an alert";
		public string confirmWithOKMessage  = "You clicked: Ok";
		public string confirmWithCancelMessage = "You clicked: Cancel";

		public JavaScriptAlertsPage(IWebDriver driver, long timeoutInSeconds) :
			base(driver, timeoutInSeconds, "http://the-internet.herokuapp.com/javascript_alerts") {
			}

		public override void open() {
			base.open();
			this.waitForElementVisible(header);
		}

	  public void clickOnJSAlertButton() {
			clickForJSAlertButton.Click();
	    	this.waitForAlertVisible();
		}

		public void clickOnJSConfirmButton() {
			clickForJSConfirmButton.Click();
	    	this.waitForAlertVisible();
		}

		public void clickOnJSPromptButton() {
			clickForJSPromptButton.Click();
	    	this.waitForAlertVisible();
		}
	}
}
